addons/product/models/product.py

```diff
@@ -598,6 +598,22 @@ def get_product_multiline_description_sale(self):
 
         return name
 
+    def _has_valid_attributes(self, valid_attribute_ids, valid_value_ids):
```

Comment:
This API is misleading: parameters are recordsets but name suggests ids!
Use `attributes` and `attribute_values`, for instance.
Same for the code inside this method and the code using it.